GVFS/GVFS.Common/FileSystem/HooksInstaller.cs

```diff
@@ -48,20 +54,15 @@ public static bool InstallHooks(GVFSContext context, out string error)
             error = string.Empty;
             try
             {
-                string installedReadObjectHookPath = Path.Combine(ExecutingDirectory, GVFSPlatform.Instance.Constants.GVFSReadObjectHookExecutableName);
-                string targetReadObjectHookPath = Path.Combine(context.Enlistment.WorkingDirectoryRoot, GVFSConstants.DotGit.Hooks.ReadObjectPath + GVFSPlatform.Instance.Constants.ExecutableExtension);
-                if (!TryHooksInstallationAction(() => CopyHook(context, installedReadObjectHookPath, targetReadObjectHookPath), out error))
-                {
-                    error = "Failed to copy " + installedReadObjectHookPath + "\n" + error;
-                    return false;
-                }
-
-                string installedVirtualFileSystemHookPath = Path.Combine(ExecutingDirectory, GVFSPlatform.Instance.Constants.GVFSVirtualFileSystemHookExecutableName);
-                string targetVirtualFileSystemHookPath = Path.Combine(context.Enlistment.WorkingDirectoryRoot, GVFSConstants.DotGit.Hooks.VirtualFileSystemPath + GVFSPlatform.Instance.Constants.ExecutableExtension);
-                if (!TryHooksInstallationAction(() => CopyHook(context, installedVirtualFileSystemHookPath, targetVirtualFileSystemHookPath), out error))
+                foreach (HookData hook in NativeHooks)

```

Comment:
why are the precommand and postcommand hooks separate?

Reply:
Pre and Post command hooks work a little differently than the others because they are **not** direct hooks that git calls.  They are using the GitHooksLoader to read a .hooks file that lists the programs to run for during pre and post command of which the pre and post commands hooks are listed.

See [TryInstallGitCommandHooks](https://github.com/Microsoft/VFSForGit/blob/3337d8643ac81f73e2a4a2e34edab986fdd3243f/GVFS/GVFS.Platform.Windows/WindowsPlatform.cs#L222) and [CreateHookCommandConfig](https://github.com/Microsoft/VFSForGit/blob/3337d8643ac81f73e2a4a2e34edab986fdd3243f/GVFS/GVFS.Platform.Windows/WindowsGitHooksInstaller.cs#L18)